instructions/steps.md

```diff
@@ -615,11 +652,14 @@ Instead we will use
 
 ```
 
-`:-` is the Elephant operator,
-or ["Update with Failure"](https://dafny.org/dafny/DafnyRef/DafnyRef#sec-update-failure)
-It will look at the return value
-and if the return is `Success` it will extract the value,
-and if the return is `Failure` it will halt and return the error.
+`:-` is the Elephant symbol,
+used here in a ["Let or Fail" expression](https://dafny.org/dafny/DafnyRef/DafnyRef#2139-let-or-fail-expression).
+You can think of this as a drastically simplified model
+of throwing exceptions.
+It will first look at the value to the right of the `:-`:
+* If it is a `Success`, it will extract the value and evaluate the expression that follows the `;`. In this case, that means the remaining lines of the function body.
+* If it is a `Failure`, it will propagate that value as the result
```

Comment:
I know they are the "same" but I think we should disambiguate
the error on the `Failure` side from the value on the `Success` side

```suggestion
* If it is a `Failure`, it will propagate that error as the result
```